Add script for EPIC deconvolution

sjspielman opened the pull request:
Closes #116 

This PR adds a script (and more!) to run `EPIC`. Here's the rundown:

- First, we observed when running `quanTIseq` that some of the gene symbols in their reference did not match ours and needed to be recoded. Therefore, I wrote a very quick notebook to look at the `EPIC` references and find any similar differences.
- I added the script to run `EPIC` and a step to call it in the analysis shell script
  - Originally I had thought to include an option for which reference to run, but `EPIC` runs _so fast_ (like, 1 second) that in the end I figured we might as well just run both at once.  
  - The script exports two items: a TSV of the fractions of interest, and an RDS file with the full EPIC objects for posterity or in case we want more diagnostic information the future. Let me know if you think one of these file exports is overkill
  - All the result files are committed too, which is most of the changes
  - My day finally came to (I think???) not unreasonably shove `purrr::pluck()` 🐔 into a script 😂 . I had fun.
- I updated READMEs all around, including a recreational change as a treat to change bullets from `-` -> `*`, which is easier moving forward for review in GitHub since `-` also means deleted line.

allyhawkins reviewed (commented):
This mostly looks good to me, I just had a few small comments. I think the biggest one is trying to understand what you might get from the RDS file that you can't capture in the TSV?

sjspielman commented on analysis/bulk-deconvolution/scripts/run-epic.R:
```diff
@@ -0,0 +1,123 @@
+
+
+# Check inputs and define paths -------
+stopifnot(
+  "A path to an input RDS file must be specified with `input_file`." = !is.null(opts$input_file),
+  "A path to an output TSV must be specified with `output_tsv_file`." = !is.null(opts$output_tsv_file),
+  "A path to an output RDS file must be specified with `output_epic_object_file`." = !is.null(opts$output_epic_object_file)
+)
+
+stopifnot("The provided `input_file` does not exist." = file.exists(opts$input_file))
```
Yes - `stopifnot()` runs all checks regardless of if an earlier one fails, so if this is indeed `NULL`, you'll just get an `object not found` error if something is null.

A minimal example:
```
> ls() # confirm empty environment
character(0)
>  stopifnot("x can't be null" = !is.null(x), "x better be 2", x == 2)
Error: object 'x' not found
```

In this example we would have hoped for just `"x can't be null"` as the error, but alas.

sjspielman commented on analysis/bulk-deconvolution/scripts/run-epic.R:
```diff
@@ -0,0 +1,123 @@
+readr::write_tsv(epic_df, opts$output_tsv_file)
+
+# full result
+readr::write_rds(epic_list, opts$output_epic_object_file)
```
I agree, this is the better move. My thought process was indeed, "hmm, _maybe_ one day we'll want the `mRNAProportions`!

sjspielman commented:
This now only exports a TSV with both of the fraction flavors that EPIC estimates, and I updated docs accordingly.
Note that I also added a tiny comment to the quanTIseq script to remind future us that it's cool that there's no seed set.

allyhawkins reviewed (commented):
Is there a way we could update this so that instead of adding a column for "matrix_name" we have a column for `mRNAProportions` and `cell_fraction` for each sample? So we would have `sample_id`, `reference`, `epic_celltype`, `cell_fraction` and `mRNAProportions`. Does that make sense?

sjspielman commented:
> So we would have sample_id, reference, epic_celltype, cell_fraction and mRNAProportions. Does that make sense?

Makes sense!

sjspielman commented:
Updated the script as requested! I also updated the `results/README.md` to explain a bit more about the columns.

allyhawkins commented on analysis/bulk-deconvolution/scripts/run-epic.R:
```diff
@@ -0,0 +1,120 @@
+    # join back into epic_df
+    dplyr::right_join(epic_df, by = c("sample_id", "epic_celltype")) |>
+    # arrange columns
+    dplyr::select(sample_id, epic_celltype, fraction, mRNAProportions)
```
We should have `reference` here too. I think you are missing a join.

sjspielman replied:
😬

allyhawkins reviewed (approved):
LGTM 🚀
